backend/bitboxbase/bitboxbase.go

```diff
@@ -126,19 +172,89 @@ func (base *BitBoxBase) ConnectElectrum() error {
 	return nil
 }
 
+// Deregister calls the backend's BitBoxBaseDeregister callback and sends a notification to the frontend, if bitboxbase is active.
+// If bitboxbase is not active, an error is returned.
+func (base *BitBoxBase) Deregister() (bool, error) {
+	if base.active {
+		// let the frontend know that the base is disconnected
+		base.fireEvent("disconnect")
+		base.onUnregister(base.bitboxBaseID)
+		base.active = false
+		return true, nil
+	}
+	return false, errp.New("Attempted call to non-active base")
+}
+
+// ChannelHash returns the bitboxbase's rpcClient noise channel hash
+func (base *BitBoxBase) ChannelHash() (string, bool) {
+	return base.rpcClient.ChannelHash()
+}
+
+// Status returns the current state of the bitboxbase.
+func (base *BitBoxBase) Status() bitboxbasestatus.Status {
+	return base.status
+}
+
+// fireEvent notifies the frontend of an event in the bitboxbase
+func (base *BitBoxBase) fireEvent(event bitboxbasestatus.Event) {
+	base.Notify(observable.Event{
+		Subject: fmt.Sprintf("/bitboxbases/%s/event", base.bitboxBaseID),
+		Action:  action.Replace,
+		Object:  event,
+	})
+}
+
+func (base *BitBoxBase) changeStatus(status bitboxbasestatus.Status) {
+	base.status = status
+	base.fireEvent(bitboxbasestatus.EventStatusChange)
+}
+
 // RPCClient returns ths current instance of the rpcClient
 func (base *BitBoxBase) RPCClient() *rpcclient.RPCClient {
 	return base.rpcClient
 }
 
 // MiddlewareInfo returns the received MiddlewareInfo packet from the rpcClient
-func (base *BitBoxBase) MiddlewareInfo() (rpcclient.SampleInfoResponse, error) {
-	response, err := base.rpcClient.SampleInfo()
-	if err != nil {
-		// intercept error so the user is not confronted with weird rpc error message
-		return response, errp.New("error received from sample info rpc call client")
+func (base *BitBoxBase) MiddlewareInfo() (rpcmessages.SampleInfoResponse, error) {
+	if !base.active {
+		err := errp.New("Attempted a call to non-active base")
+		return rpcmessages.SampleInfoResponse{}, err
+	}
+	return base.rpcClient.GetSampleInfo()
+}
+
+// VerificationProgress returns the received VerificationProgress packet from the rpcClient
+func (base *BitBoxBase) VerificationProgress() (rpcmessages.VerificationProgressResponse, error) {
+	if !base.active {
+		err := errp.New("Attempted a call to non-active base")
+		return rpcmessages.VerificationProgressResponse{}, err
+	}
+	return base.rpcClient.GetVerificationProgress()
+}
+
+// SyncOption returns true if the chosen sync option was executed successfully
+func (base *BitBoxBase) SyncOption(option string) (bool, error) {
```

Comment:
avoid stringly types. Make it a proper type with const values, then every option can be documented and there is less room for silent errors.

Similar to https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/backend/accounts/event.go